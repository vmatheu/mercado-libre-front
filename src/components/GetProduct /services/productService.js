import axios from 'axios';
import { PRODUCT_SERVICE_ENDPOINT_ID } from '../../constants';

const transformProductSearch = (respuesta) => ({
    id: respuesta.item.id,
    title: respuesta.item.title,
    currency: respuesta.item.price.currency,
    amount: respuesta.item.price.amount,
    decimals: respuesta.item.price.decimals,
    picture: respuesta.item.picture,
    condition: respuesta.item.condition,
    freeShipping: respuesta.item.freeShipping,
    description: respuesta.item.description
})



export const findProductById = async (search) => {
  const productServiceEnpoint = PRODUCT_SERVICE_ENDPOINT_ID;
  const data = await axios.get(`${productServiceEnpoint}/${search}`)
    .then(response => ({
      item: transformProductSearch(response.data),
      categories: response.data.categories,
    }))
    .catch(error => {
      console.error(error)
      return []
    });
  return data;
};

export default { findProductById };
