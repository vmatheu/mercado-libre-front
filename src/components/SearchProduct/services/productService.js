import axios from 'axios';
import { PRODUCT_SERVICE_ENDPOINT } from '../constants';

export const findProductBySearch = async (search) => {
  const productServiceEnpoint = PRODUCT_SERVICE_ENDPOINT['development'].url;
  const data = await axios.get(`${productServiceEnpoint}${search}`)
    .then(response => (response.data))
    .catch(error => ([]));
  return data;
};

export default { findProductBySearch };
