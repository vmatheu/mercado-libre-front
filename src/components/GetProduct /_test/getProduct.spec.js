import React from 'react';
import toJson from 'enzyme-to-json';
import reactRouterDom from "react-router-dom";
import { shallow } from 'enzyme';
import productService from '../services/productService';
import { GetProduct, onClick } from '..';

jest.mock('../services/productService');
jest.mock('react-router-dom');

describe('<GetProduct/>', () => {

  beforeEach(() => {
    reactRouterDom.useParams = jest.fn();
    reactRouterDom.useParams.mockReturnValue({
      searchParam: "La busqueda"
    });
  });

  describe('With search product by string query', () => {
    it('should have the same ui structure', () => {
      const wrapper = shallow(<GetProduct />);
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
    });
  });
});
