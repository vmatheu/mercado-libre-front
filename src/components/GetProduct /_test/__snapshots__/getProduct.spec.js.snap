// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GetProduct/> With search product by string query should have the same ui structure 1`] = `<div />`;
